// @web-tools/ui/src/index.ts
// ── Components ──────────────────────────────────────────────────────────────
export { AppHeader } from './components/AppHeader'
export { Button } from './components/Button'
export { CopyButton } from './components/CopyButton'
export { EmptyState } from './components/EmptyState'
export { ErrorBanner } from './components/ErrorBanner'
export { ModeSelector } from './components/ModeSelector'
export { ShortcutHints, getPlatformMod, getPlatformShift } from './components/ShortcutHints'
export { TextArea } from './components/TextArea'

// ── Icons ────────────────────────────────────────────────────────────────────
export { CopyIcon, HomeIcon } from './components/icons'

// ── Hooks ────────────────────────────────────────────────────────────────────
export { useCopyToClipboard } from './hooks/useCopyToClipboard'
export { useKeyboardShortcut } from './hooks/useKeyboardShortcut'
export { useStorage } from './hooks/useStorage'

// ── Utils ────────────────────────────────────────────────────────────────────
export { formatBytes } from './utils/formatBytes'
export { isMacPlatform, modKey, shiftKey } from './utils/platform'
